### PR TITLE
Add autopull service for testing & pre-commit-hooks

### PR DESCRIPTION
**Pull request by ahuston-0**
Adds an service to pull git repositories (currently setup as a template service in systemd, but not in nix) and allows the service to then trigger the nixos-upgrade service. 

Adds nixpkgs.fmt to the flake config so we can run `nix fmt`

Adds pre-commit-hooks

Signed-off-by: ahuston-0 <aliceghuston@gmail.com>

### CODE REVIEW

**Comment by ahuston-0** [comment]
Switched from (Cachix's pre-commit-hooks)[https://github.com/cachix/pre-commit-hooks.nix] to (nix-pre-commit)[https://github.com/jmgilman/nix-pre-commit] as the issue I was having with the `sops-nix` hook conflicting doesn't seem to happen there. Currently there is one hook for running `nix fmt` and another for `nix flake check`. Please let me know if this needs more hooks before going to main

File: flake.nix [hunk 1]
```diff
@@ -26,16 +26,46 @@
         nixpkgs-stable.follows = "nixpkgs";
       };
     };
+    # pre-commit-hooks ={
+    #   url = "github:cachix/pre-commit-hooks.nix";
+    #   # below doesnt seem to work as expected...
+    #   # inputs.nixpkgs.follow = "nixpkgs";
+    # };
+    nix-pre-commit.url = "github:jmgilman/nix-pre-commit";
```
Comment (DerDennisOP) [inline flake.nix:34]: check with `nix flake metadata`

Comment (DerDennisOP) [inline flake.nix:34]:
```suggestion
    nix-pre-commit = {
      url = "github:jmgilman/nix-pre-commit";
      inputs = {
        nixpkgs.follow = "nixpkgs";
        flake-utils.follow = "flake-utils";
      };
    };
```
try that

**Review by DerDennisOP — CHANGES_REQUESTED** [review]
everything else looks good

File: systems/palatine-hill/configuration.nix [hunk 2]
```diff
@@ -47,6 +47,11 @@
     };
 
     openssh.ports = [ 666 ];
+    autopull = {
```
[the remark this answers is not in the document]
Comment (ahuston-0) [reply systems/palatine-hill/configuration.nix:50]: as discussed, moving this out.

[the remark this answers is not in the document]
Comment (ahuston-0) [reply systems/palatine-hill/configuration.nix]: done with [12500be](https://github.com/RAD-Development/nix-dotfiles/pull/4/commits/12500bee1499961a626862260ac79833e6c1ef75)

**Review by DerDennisOP — APPROVED** [review]

**Pull request merged**